### PR TITLE
Re-enable dynamo tests with TODOs

### PR DESCRIPTION
**Pull request by wonjoolee95**
Re-enable dynamo tests with TODOs

---

Had a quick sync with Jack earlier today regarding the failing dynamo tests. Given that we want to prevent new possible regressions and that only `CompileTime` metrics differ, we're re-enabling these tests for now. 

I'll keep https://github.com/pytorch/xla/issues/4680 open and post findings/updates there.

### CODE REVIEW

File: test/dynamo/test_dynamo.py [hunk 1]
```diff
@@ -267,9 +266,11 @@ def test_resnet18(self):
     # Graph 3: optimizer
     # Graph 4: sync input for backward
     # Graph 5: sync input for backward (TODO(JackCaoG) understand why there are two graphs)
-    self.assertEqual(met.metric_data('CompileTime')[0], 5)
+    # TODO @wonjoo CompileTime has increased 5->7 after enabling functionalization (#4680)
+    self.assertEqual(met.metric_data('CompileTime')[0], 7)
     # We execute 4 grphs per step when optimizer is enabled.
-    self.assertEqual(met.metric_data('ExecuteTime')[0], sample_count * 4)
+    # TODO @wonjoo ExecuteTime has increased by 1 after enabling functionalization (#4680)
+    self.assertEqual(met.metric_data('ExecuteTime')[0], (sample_count * 4) + 1)
```
Comment (wonjoolee95) [inline test/dynamo/test_dynamo.py:273]: @JackCaoG, I just saw that this was failing as well -- the `ExecuteTime` is offset by 1 regardless of the sample size.

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Comment by JackCaoG** [comment]
@wonjoolee95 since you are at it, can you fix the random seed for each test? This way test doesn't fail randomly. @yeounoh FYI

File: test/dynamo/test_dynamo.py [hunk 2]
```diff
@@ -14,6 +14,10 @@
 
 class DynamoInferenceBasicTest(unittest.TestCase):
 
+  @classmethod
+  def setUpClass(self):
+    torch.manual_seed(42)
```
Comment (wonjoolee95) [inline test/dynamo/test_dynamo.py:19]: We probably don't need the seed for the inference test? But just added for sake of completeness.

**Comment by wonjoolee95** [comment]
> @wonjoolee95 since you are at it, can you set a fixed random seed for each test? This way test doesn't fail randomly. @yeounoh FYI

Sure thing, added a @classmethod:
```
  @classmethod
  def setUpClass(self):
    torch.manual_seed(42)
```

**Comment by wonjoolee95** [comment]
The CPU CI is green but GPU test failed with:
```
======================================================================
FAIL: test_optimizer (__main__.TestSyncFreeSGD)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_syncfree_optimizers.py", line 127, in test_optimizer
    self._test_optimizer(syncfree.SGD, torch.optim.SGD, {
  File "test/test_syncfree_optimizers.py", line 103, in _test_optimizer
    np.testing.assert_allclose(
  File "/opt/conda/lib/python3.8/site-packages/numpy/testing/_private/utils.py", line 1527, in assert_allclose
    assert_array_compare(compare, actual, desired, err_msg=str(err_msg),
  File "/opt/conda/lib/python3.8/site-packages/numpy/testing/_private/utils.py", line 844, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=0.01, atol=0.01

Mismatched elements: 13 / 16000 (0.0813%)
Max absolute difference: 0.09543511
Max relative difference: 5.716095
 x: array([[-0.005069,  0.052122, -0.037159, ...,  0.038525, -0.04917 ,
         0.036206],
       [-0.03475 , -0.038435, -0.024201, ...,  0.001684,  0.051582,...
 y: array([[-0.005069,  0.052122, -0.037159, ...,  0.038525, -0.04917 ,
         0.036206],
       [-0.03475 , -0.038435, -0.024201, ...,  0.001684,  0.051582,...

----------------------------------------------------------------------
```
Doesn't seem related to this PR since all this PR does is enable two of the dynamo unit tests? Let me rebase and re-run.

**Comment by wonjoolee95** [comment]
Ok CIs are green after rebasing, merging this.

**Pull request merged**